### PR TITLE
feature(license): check license before node start

### PR DESCRIPTION
**Pull request by savonarola**
Check license before node starts so that it does not crash.

### CODE REVIEW

File: rebar.config.erl [hunk 1]
```diff
@@ -239,10 +239,12 @@ overlay_vars_rel(RelType) ->
 
 overlay_vars_edition(ce) ->
     [ {emqx_schema_mod, emqx_conf_schema}
+    , {is_enterprise, "no"}
     , {emqx_machine_boot_apps, emqx_machine_boot_app_list(ce)}
     ];
 overlay_vars_edition(ee) ->
     [ {emqx_schema_mod, emqx_enterprise_conf_schema}
+    , {is_enterprise, "yes"}
```
Comment (thalesmg) [inline rebar.config.erl:247]: Just a reminder that we'll also need this variable set in `mix.exs`

**Review by thalesmg — APPROVED** [review]

File: bin/nodetool [hunk 2]
```diff
@@ -253,6 +257,15 @@ chkconfig(File) ->
             halt(1)
     end.
 
+check_license(Config) ->
+    ok = application:load(emqx_license),
+    case emqx_license:read_license(Config) of
+        {ok, _} -> ok;
```
Comment (zmstone) [inline bin/nodetool:263]: Should check license validity too?

Comment (zmstone) [reply bin/nodetool:263]: Ok. Please add this as a comment.

**Review by zhongwencool — APPROVED** [review]

**Review by zmstone — CHANGES_REQUESTED** [review]
Add comments to clarify at which level of validity check is done.

**Pull request merged**